### PR TITLE
fix: state read dry run sets gas price to 0

### PR DESCRIPTION
**Pull request by MujkicA**
### Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have updated the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary labels.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by Br1ght0ne — APPROVED** [review]
Great regression test, LGTM.

File: packages/fuels-programs/src/calls/call_handler.rs [hunk 1]
```diff
@@ -198,7 +198,7 @@ where
                 .build(provider)
                 .await?;
 
-            provider.dry_run_opt(tx, false, None).await?
+            provider.dry_run_opt(tx, false, Some(0)).await?
```
Comment (digorithm) [inline packages/fuels-programs/src/calls/call_handler.rs:201]: Can you clarify the difference between none and some(0)?

**Review by digorithm — APPROVED** [review]
Thanks for tackling this so quickly!

**Review by segfault-magnet — APPROVED** [review]

**Review by hal3e — APPROVED** [review]

File: packages/fuels-programs/src/calls/call_handler.rs (hunk 1 above)
Comment (MujkicA) [reply packages/fuels-programs/src/calls/call_handler.rs:201]: We're telling the node to assume that the gas price is 0 for this dry run. `None` would mean it uses whatever the current price is.

**Pull request merged**